File: Codigo/MobileApp/lib/app/data/models/address.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'address.freezed.dart';
part 'address.g.dart';

@Freezed()
class Address with _$Address {
  const Address._();

  const factory Address({
    required int addressId,
    required String cityName,
    required String countryState,
    required String streetName,
    required String streetNumber,
    required String? unitNumber,
    required String postalCode,
    required String neighborhoodName,
    required double lat,
    required double lng,
  }) = _Address;

  factory Address.fromJson(Map<String, dynamic> json) =>
      _$AddressFromJson(json);

  String get formatted =>
      '$streetName, $streetNumber${unitNumber != null ? '/$unitNumber' : ''} - $neighborhoodName, $cityName - $countryState';
}

File: Codigo/DeliveryApi/src/classes/dataclasses/Address.py
from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class Address:

    city_name: str
    country_state: str
    street_name: str
    street_number: str
    postal_code: str
    neighborhood_name: str
    unit_number: Optional[str] = None

    def __str__(self) -> str:
        return f'''{
            self.street_name
            } {self.street_number} {self.neighborhood_name} {self.city_name} {self.country_state} {self.postal_code}'''
